pandas/tests/indexing/multiindex/test_loc.py

```diff
@@ -379,6 +379,22 @@ def test_multiindex_setitem_columns_enlarging(self, indexer, exp_value):
         )
         tm.assert_frame_equal(df, expected)
 
+    def test_sorted_multiindex_after_union(self):
```

Comment:
are all 3 cases from the OP represented (elsewhere)?

Reply:
Good point, added the working case too